pkg/internal/token/adalclientcertcredential_test.go

```diff
@@ -0,0 +1,99 @@
+package token
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewADALClientCertCredential(t *testing.T) {
+	testCases := []struct {
+		name           string
+		opts           *Options
+		expectError    bool
+		expectErrorMsg string
+		expectNil      bool
+		expectName     string
+	}{
+		{
+			name: "valid options",
+			opts: &Options{
+				ClientID:           "test-client-id",
+				TenantID:           "test-tenant-id",
+				ClientCert:         "test-cert-path",
+				ClientCertPassword: "test-cert-password",
+				IsLegacy:           true,
+			},
+			expectError: false,
+			expectNil:   false,
+			expectName:  "ADALClientCertCredential",
```

Comment:
What's the utility of testing this?

```go
func (c *ADALClientCertCredential) Name() string {
	return "ADALClientCertCredential"
}
```

Reply:
(and not testing the fields in the returned `ADALClientCertCredential`, for instance? To whatever level we even want to test a constructor like `newADALClientCertCredential`, this set of tests allows an implementation that validates input options but doesn't actually create any output to pass, which seems like the only thing we would want to test ;) )

Reply:
yeah, these tests are just for the sake of coverage. most of the credential flows based on track2 have the output/behavior validation. Not this one, though. Too lazy to do this for adal which is not supported anymore :P